agent/engine/docker_task_engine.go

```diff
@@ -553,7 +553,23 @@ func (engine *DockerTaskEngine) pullAndUpdateContainerReference(task *api.Task,
 		return DockerContainerMetadata{Error: TaskStoppedBeforePullBeginError{task.Arn}}
 	}
 
-	metadata := engine.client.PullImage(container.Image, container.RegistryAuthentication)
+	var metadata DockerContainerMetadata
+	imagePullBackoff := utils.NewSimpleBackoff(250*time.Millisecond, time.Second, 0.20, 1.5)
+	_ = utils.RetryNWithBackoff(imagePullBackoff, 10, func() error {
+		metadata = engine.client.PullImage(container.Image, container.RegistryAuthentication)
+		if container.IsInternal() {
+			// "internal" containers are created by the agent and we don't expect to actually pull them
+			return nil
+		}
+		if imagePullError, ok := metadata.Error.(CannotPullContainerError); ok {
+			seelog.Warnf("When pulling %s: %s", container.Image, pullError)
```

Comment:
`pullError` and `imagePullError` aren't supposed to be distinct names. Can we fix this commit before the PR is merged? Either name is fine to use.

(I realize that this code doesn't exist as written in the final PR, but I do want bisects, etc, to work and we shouldn't have unbuildable code in the repo.)

Reply:
Yeah, I'll fix this.